autonomy/chain/metadata.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2023 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Metadata helpers."""
+
+import json
+from collections import OrderedDict
+from pathlib import Path
+
+from aea.configurations.data_types import PublicId
+from aea.helpers.cid import CID
+from aea.helpers.ipfs.base import IPFSHashOnly
+from aea_cli_ipfs.ipfs_utils import IPFSTool
+
+
+IPFS_URI_PREFIX = "ipfs://"
+BASE16_HASH_PREFIX = "f01701220"
+CONFIG_HASH_STRING_PREFIX = "0x"
+UNIT_HASH_PREFIX = CONFIG_HASH_STRING_PREFIX + "{metadata_hash}"
+
+
+def serialize_metadata(
+    package_hash: str,
+    public_id: PublicId,
+    description: str,
+    nft_image_hash: str,
+) -> str:
+    """Serialize metadata."""
+    metadata = OrderedDict(
+        {
+            "name": f"{public_id.author}/{public_id.name}",
+            "description": description,
+            "code_uri": f"{IPFS_URI_PREFIX}{package_hash}",
+            "image": f"ipfs://{nft_image_hash}",
+            "attributes": [{"trait_type": "version", "value": str(public_id.version)}],
+        }
+    )
+    return json.dumps(obj=metadata)
+
+
+def publish_metadata(
```

Comment:
Same as above

Reply:
Do we have tests which show that this produces 1:1 the same results as the app?

Reply:
I've added the tests for the helper functions in `autonomy.chain` module but they reside in a single test file. I'll restructure the tests once we're done with the functionality